### PR TITLE
Preview target rebuild in engine instead of http call

### PR DESCRIPTION
**Pull request by phuongnq**
### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/6223

### CODE REVIEW

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please also add to the Deployer Upgrade Manager (UM) to automatically remove it from existing installations.

**Comment by phuongnq** [comment]
@sumerjabri I added the removal processor to UM. It will match for the same processor with the same properties for removal.

File: src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    /**
+     * Check if the processor has all the required properties
+     * @param processorObj processor object
+     * @return true if all properties are the same, false otherwise
+     */
+    protected boolean hasAllProperties(Map<String, Object> processorObj) {
+        if (processorObj.get(PROCESSOR_NAME_CONFIG_KEY).equals(processorName)) {
```
Comment (jmendeza) [inline src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java]: This condition can be inverted to reduce indentation

File: src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import org.craftercms.commons.config.ConfigurationException;
+import org.craftercms.deployer.api.Target;
+
+import java.util.*;
+
+import static org.craftercms.commons.config.ConfigUtils.getRequiredStringProperty;
+import static org.craftercms.deployer.impl.DeploymentConstants.PROCESSOR_NAME_CONFIG_KEY;
+
+/**
+ * Operation to remove a new processor to the pipeline of the target
```
Comment (jmendeza) [inline src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java]: "Operation to remove a processor from the pipeline of the target"

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java (hunk 1 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java]: I updated this. Thank you!

File: src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java (hunk 2 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/deployer/impl/upgrade/operations/RemoveProcessorUpgradeOperation.java]: I updated this. Thank you!

**Review by jmendeza — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**